app/routes.py

```diff
@@ -1,2 +1,251 @@
+videos_bp = Blueprint("videos", __name__, url_prefix="/videos")
+rentals_bp = Blueprint("rentals", __name__, url_prefix="/rentals")
+
+
+#helper functions
+def valid_customer_input(form_data):
+    if "name" not in form_data:
+        abort(make_response({"error": "Missing customer name"},400)) 
+    if "postal_code" not in form_data:
+        abort(make_response({"error": "Missing customer postal_code"},400))
+    if "phone" not in form_data:
+        abort(make_response({"error": "Missing customer phone"},400))
+    return True
+
+
+def valid_video_input(form_data):
+    if "title" not in form_data:
+        abort(make_response({"error": "Missing video title"},400))
+    if "release_date" not in form_data:
+        abort(make_response({"error": "Missing video release_date"},400))
+    if "total_inventory" not in form_data:
+        abort(make_response({"error": "Missing video total_inventory"},400))
+    return True
+
+#routes
+@customers_bp.route("", methods=["GET"])
+def get_customers():
+    customers = Customer.query.all()
+    customers_response = [customer.to_dict() for customer in customers]
+    return jsonify(customers_response)
+
+
+@customers_bp.route("/<customer_id>", methods=["GET"])
+def get_one_customer(customer_id):
+    customer = Customer.query.get(customer_id)
+    if not customer:
+        return ({"error": f"No customer with ID #{customer_id}"}, 404)
+    return customer.to_dict()
+
+
+@customers_bp.route("", methods=["POST"])
+def create_customer():
+    request_body = request.get_json()
+    valid_customer_input(request_body)
+    new_customer = Customer(name = request_body["name"],
+                    postal_code = request_body["postal_code"],
+                    phone = request_body["phone"])
+    new_customer.registered_at = datetime.utcnow()
+    db.session.add(new_customer)
+    db.session.commit()
+    return ({"id": new_customer.customer_id}, 201) 
+
+
+@customers_bp.route("/<customer_id>", methods=["PUT"])
+def update_customer(customer_id):
+    customer = Customer.query.get(customer_id)
+    if not customer:
+        return ({"error": f"No customer with ID #{customer_id}"}, 404)
+    form_data = request.get_json()
+    valid_customer_input(form_data)
+    customer.name = form_data["name"]
+    customer.postal_code = form_data["postal_code"]
+    customer.phone = form_data["phone"]
+    db.session.commit()
+    return customer.to_dict()
+    
+
+@customers_bp.route("/<customer_id>", methods=["DELETE"])
+def delete_customer(customer_id):
+    customer = Customer.query.get(customer_id)
+    if not customer:
+        return ({"error": f"No customer with ID #{customer_id}"}, 404)
+    db.session.delete(customer)
+    db.session.commit()
+    return make_response({"id": customer.customer_id}, 200)
+
+
+@videos_bp.route("", methods=["GET"])
+def get_videos():
+    videos = Video.query.all()
+    videos_response = [video.to_dict() for video in videos]
+    return jsonify(videos_response)
```

Comment:
I like the list comprehension.  Since you only use `videos_response` once you can "inline" it:

```suggestion
    return jsonify([video.to_dict() for video in videos])
```